transforms_stats#"Test Get Transform Stats"  fails because pages_processed/checkpoint don't match

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3614/

Reproduction:

```
./gradlew :x-pack:plugin:integTestRunner --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_frame/transforms_stats/Test get transform stats}" -Dtests.seed=5B30818AA97DC12A -Dtests.security.manager=true -Dtests.locale=en-KE -Dtests.timezone=Europe/Sarajevo -Dcompiler.java=12 -Druntime.java=11 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
```

This failed on CI with:

> java.lang.AssertionError: Failure at [data_frame/transforms_stats:53]: transforms.0.stats.pages_processed didn't match expected value:
transforms.0.stats.pages_processed: expected [0] but was [1]

It also fails for me locally, but with a different part of the test failing (checkpoint instead of the pages_processed).  Fails reliably for me

> Caused by:
        java.lang.AssertionError: transforms.0.state.checkpoint didn't match expected value:
         transforms.0.state.checkpoint: expected [0] but was [1]

```
Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "count" : 1,
  1>       "transforms" : [
  1>         {
  1>           "id" : "airline-transform-stats",
  1>           "state" : {
  1>             "task_state" : "started",
  1>             "indexer_state" : "started",
  1>             "checkpoint" : 1,
  1>             "progress" : {
  1>               "total_docs" : 0,
  1>               "docs_remaining" : 0,
  1>               "percent_complete" : 100.0
  1>             }
  1>           },
  1>           "stats" : {
  1>             "pages_processed" : 1,
  1>             "documents_processed" : 0,
  1>             "documents_indexed" : 0,
  1>             "trigger_count" : 1,
  1>             "index_time_in_ms" : 0,
  1>             "index_total" : 0,
  1>             "index_failures" : 0,
  1>             "search_time_in_ms" : 8,
  1>             "search_total" : 1,
  1>             "search_failures" : 0
  1>           },
  1>           "checkpointing" : {
  1>             "in_progress" : {
  1>               "timestamp_millis" : 1558460091734
  1>             },
  1>             "operations_behind" : -1
  1>           }
  1>         }
  1>       ]
  1>     }
  1>   }
  1> }]
```

Looks like it started failing yesterday, so I'm guessing it's somehow related to https://github.com/elastic/elasticsearch/pull/42067 /cc @davidkyle 

![image](https://user-images.githubusercontent.com/1224228/58118054-0e9a8d80-7bce-11e9-96c6-691c761efdbd.png)

Since it fails reliably I'm going to mute this.

Reset to current master, cleaned, re-ran test.  Looks like it still fails :(

Happy to hand this over if you want to handle it, muting or not, etc :)

```
$ git rev-parse HEAD
be412ca83f0d4644712a7baa8a6330b1e72b2858
```

```
org.elasticsearch.xpack.test.rest.XPackRestIT > test {p0=data_frame/transforms_stats/Test get transform stats} FAILED
    java.lang.AssertionError: Failure at [data_frame/transforms_stats:52]: transforms.0.state.checkpoint didn't match expected value:
     transforms.0.state.checkpoint: expected [0] but was [1]

        Caused by:
        java.lang.AssertionError: transforms.0.state.checkpoint didn't match expected value:
         transforms.0.state.checkpoint: expected [0] but was [1]
REPRODUCE WITH: ./gradlew :x-pack:plugin:integTestRunner --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_frame/transforms_stats/Test get transform stats}" -Dtests.seed=5B30818AA97DC12A -Dtests.security.manager=true -Dtests.locale=en-KE -Dtests.timezone=Europe/Sarajevo -Dcompiler.java=12 -Druntime.java=11 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*

Suite: Test class org.elasticsearch.xpack.test.rest.XPackRestIT
  1> [2019-05-21T19:54:25,615][INFO ][o.e.x.t.r.XPackRestIT    ] [test] [p0=data_frame/transforms_stats/Test get transform stats] before test
  1> [2019-05-21T19:54:25,775][INFO ][o.e.x.t.r.XPackRestIT    ] [test] initializing REST clients against [http://[::1]:47477]
  1> [2019-05-21T19:54:26,443][INFO ][o.e.x.t.r.XPackRestIT    ] [test] initializing client, minimum es version [8.0.0], master version, [8.0.0], hosts [http://[::1]:47477]
  1> [2019-05-21T19:54:33,568][INFO ][o.e.x.t.r.XPackRestIT    ] [test] Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "count" : 1,
  1>       "transforms" : [
  1>         {
  1>           "id" : "airline-transform-stats",
  1>           "state" : {
  1>             "task_state" : "started",
  1>             "indexer_state" : "started",
  1>             "checkpoint" : 1,
  1>             "progress" : {
  1>               "total_docs" : 0,
  1>               "docs_remaining" : 0,
  1>               "percent_complete" : 100.0
  1>             }
  1>           },
  1>           "stats" : {
  1>             "pages_processed" : 1,
  1>             "documents_processed" : 0,
  1>             "documents_indexed" : 0,
  1>             "trigger_count" : 1,
  1>             "index_time_in_ms" : 0,
  1>             "index_total" : 0,
  1>             "index_failures" : 0,
  1>             "search_time_in_ms" : 8,
  1>             "search_total" : 1,
  1>             "search_failures" : 0
  1>           },
  1>           "checkpointing" : {
  1>             "current" : {
  1>               "timestamp_millis" : 1558461273424
  1>             },
  1>             "operations_behind" : 0
  1>           }
  1>         }
  1>       ]
  1>     }
  1>   }
  1> }]
```

Also muting `"Test get multiple transform stats where one does not have a task"` which is failing with a related set of errors

```
java.lang.AssertionError: Failure at [data_frame/transforms_stats:166]: transforms.0.state.indexer_state didn't match expected value:
transforms.0.state.indexer_state: expected [started] but was [indexing]
```

```
 Stash dump on test failure [{
14:52:46   1>   "stash" : {
14:52:46   1>     "body" : {
14:52:46   1>       "count" : 2,
14:52:46   1>       "transforms" : [
14:52:46   1>         {
14:52:46   1>           "id" : "airline-transform-stats",
14:52:46   1>           "state" : {
14:52:46   1>             "task_state" : "started",
14:52:46   1>             "indexer_state" : "indexing",
14:52:46   1>             "checkpoint" : 1,
14:52:46   1>             "progress" : {
14:52:46   1>               "total_docs" : 0,
14:52:46   1>               "docs_remaining" : 0,
14:52:46   1>               "percent_complete" : 100.0
14:52:46   1>             }
14:52:46   1>           },
14:52:46   1>           "stats" : {
14:52:46   1>             "pages_processed" : 1,
14:52:46   1>             "documents_processed" : 0,
14:52:46   1>             "documents_indexed" : 0,
14:52:46   1>             "trigger_count" : 1,
14:52:46   1>             "index_time_in_ms" : 0,
14:52:46   1>             "index_total" : 0,
14:52:46   1>             "index_failures" : 0,
14:52:46   1>             "search_time_in_ms" : 2,
14:52:46   1>             "search_total" : 1,
14:52:46   1>             "search_failures" : 0
14:52:46   1>           },
14:52:46   1>           "checkpointing" : {
14:52:46   1>             "in_progress" : {
14:52:46   1>               "timestamp_millis" : 1558464256316
14:52:46   1>             },
14:52:46   1>             "operations_behind" : -1
14:52:46   1>           }
14:52:46   1>         },
14:52:46   1>         {
14:52:46   1>           "id" : "airline-transform-stats-dos",
14:52:46   1>           "state" : {
14:52:46   1>             "task_state" : "stopped",
14:52:46   1>             "indexer_state" : "stopped",
14:52:46   1>             "checkpoint" : 0
14:52:46   1>           },
14:52:46   1>           "stats" : {
14:52:46   1>             "pages_processed" : 0,
14:52:46   1>             "documents_processed" : 0,
14:52:46   1>             "documents_indexed" : 0,
14:52:46   1>             "trigger_count" : 0,
14:52:46   1>             "index_time_in_ms" : 0,
14:52:46   1>             "index_total" : 0,
14:52:46   1>             "index_failures" : 0,
14:52:46   1>             "search_time_in_ms" : 0,
14:52:46   1>             "search_total" : 0,
14:52:46   1>             "search_failures" : 0
14:52:46   1>           },
14:52:46   1>           "checkpointing" : {
14:52:46   1>             "operations_behind" : 0
14:52:46   1>           }
14:52:46   1>         }
14:52:46   1>       ]
14:52:46   1>     }
14:52:46   1>   }
14:52:46   1> }]
```

```
./gradlew :x-pack:plugin:integTestRunner --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_frame/transforms_stats/Test get multiple transform stats where one does not have a task}" -Dtests.seed=7856C8AD2A4F2538 -Dtests.security.manager=true -Dtests.locale=hsb -Dtests.timezone=Africa/Malabo -Dcompiler.java=12 -Druntime.java=12 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
```

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
@@ -49,7 +49,7 @@ teardown:
   - match: { transforms.0.id: "airline-transform-stats" }
   - match: { transforms.0.state.indexer_state: "/started|indexing/" }
   - match: { transforms.0.state.task_state: "started" }
-  - match: { transforms.0.state.checkpoint: 0 }
+  - lte: { transforms.0.state.checkpoint: 1 }
   - lte: { transforms.0.stats.pages_processed: 1 }
   - match: { transforms.0.stats.documents_processed: 0 }
   - match: { transforms.0.stats.documents_indexed: 0 }
@@ -163,7 +163,7 @@ teardown:
         transform_id: "*"
   - match: { count: 2 }
   - match: { transforms.0.id: "airline-transform-stats" }
-  - match: { transforms.0.state.indexer_state: "started" }
+  - match: { transforms.0.state.indexer_state: "/started|indexing/" }
   - match: { transforms.1.id: "airline-transform-stats-dos" }
   - match: { transforms.1.state.indexer_state: "stopped" }
 

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
@@ -42,9 +42,6 @@ teardown:
 
 ---
 "Test get transform stats":
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/42309"
   - do:
       data_frame.get_data_frame_transform_stats:
         transform_id: "airline-transform-stats"

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/transforms_stats.yml
@@ -149,9 +149,6 @@ teardown:
 
 ---
 "Test get multiple transform stats where one does not have a task":
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/42309"
   - do:
       data_frame.put_data_frame_transform:
         transform_id: "airline-transform-stats-dos"

